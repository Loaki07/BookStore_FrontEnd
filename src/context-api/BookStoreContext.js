import React, { useState, createContext } from 'react';

export const BookStoreContext = createContext();

export const BookStoreProvider = (props) => {
  const [state, setState] = useState({
    loginUserEmailId: '',
    loginUserPassword: '',
    toggleLogInSignUp: false,
    message: 'baseState',
  });

  return (
    <div>
      <BookStoreContext.Provider value={[state, setState]}>
        {props.children}
      </BookStoreContext.Provider>
    </div>
  );
};
