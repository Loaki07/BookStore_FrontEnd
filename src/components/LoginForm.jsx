import React, {useContext} from 'react'
import { BookStoreContext } from '../context-api/BookStoreContext'
import { Formik, Form as FormikForm, Field, ErrorMessage } from 'formik'
import * as Yup from 'yup'
import FormErrorMessage from './FormErrorMessage.jsx';
import { Form, Button } from 'react-bootstrap';
import UserApis from "../services/UserApis"
import 'bootstrap/dist/css/bootstrap.min.css';
import {toast} from 'react-toastify'
import 'react-toastify/dist/ReactToastify.min.css'
toast.configure()

const { logIn } = new UserApis();

const initialValues = {
  emailId: "",
  password: "",
}

const validationSchema = Yup.object().shape({
  emailId: Yup.string()
              .email("Invalid email format")
              .required("Required!"),
  password: Yup.string()
               .min(6, 'Minimum 6 characters required!')
               .required('Required!'),
})

const LoginForm = ({ history }) => {
  const [state, setState] = useContext(BookStoreContext);

  const handleFormChange = (e) =>{
    setState({ ...state, [e.target.name]: e.target.value})
  }

  const handleFormSubmit = async () => {
    try {
      console.log("From Submit", state);
      const loginUserObject = {
        emailId: state.emailId,
        password: state.password
      }
      const result = await logIn(loginUserObject);
      if (result.status === 200) {
        toast.success('Login Successfull !', {
          position: toast.POSITION.TOP_CENTER
        })
        localStorage.setItem('token', result.data.token);
        // props.history.push('/dashboard');
        console.log(result);
      }
    } catch (error) {
      toast.error(error.message, {
        position: toast.POSITION.TOP_CENTER
      });
    }
  } 

  return (
    <Formik 
      initialValues={initialValues}
      validationSchema={validationSchema}
      onSubmit={handleFormSubmit}
      >
        {({ handleSubmit, handleChange, values, handleBlur }) => (
          <FormikForm>
            <Form.Group>
              <Form.Label>Email Id</Form.Label>
              <Field 
                  type="email" 
                  placeholder="Email Id" 
                  className="form-control"
                  name="emailId" 
                  value={values.emailId} 
                  onInput={handleChange}
                  onChange={handleFormChange} 
                  autoComplete="off" 
                ></Field>
                <ErrorMessage 
                  name="emailId" 
                  component={FormErrorMessage} 
                ></ErrorMessage>
            </Form.Group>
            <Form.Group>
            <Form.Label>Password</Form.Label>
              <Field 
                    type="password" 
                    placeholder="Password" 
                    className="form-control"
                    name="password" 
                    value={values.password} 
                    onInput={handleChange}
                    onChange={handleFormChange} 
                    autoComplete="off" 
                  ></Field>
                  <ErrorMessage 
                  name="password" 
                  component={FormErrorMessage} 
                ></ErrorMessage>
            </Form.Group>
            <Form.Group>
              <Button
                variant="secondary"
                type='button'
                size="sm"
              >Forgot Password?</Button>
            </Form.Group>
            <Form.Group>
              <Button 
                variant="danger"
                type='submit'
                onSubmit={handleSubmit}
                block
              >Login</Button>
            </Form.Group>
            <Form.Group>
              <span>or</span>
            </Form.Group>
            <Form.Row>
              <Form.Group>
                <Button 
                  variant="info"
                  type='button'
                >
                  FaceBook
                </Button>
                <Button 
                  variant="secondary"
                  type='button'
                >
                  Google
                </Button>
              </Form.Group>
            </Form.Row>
          </FormikForm>
        )}
    </Formik>
  )
}

export default LoginForm
