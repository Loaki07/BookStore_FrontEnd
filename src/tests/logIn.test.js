import React, { useContext, useState } from 'react';
import { configure, mount, shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import { BookStoreProvider, BookStoreContext, useBookStoreContext } from '../context-api/BookStoreContext';
import LoginForm from '../components/LoginForm.jsx';

configure({ adapter: new Adapter() });

describe('Login', () => {
  const wrapper = mount(
    <BookStoreProvider>
      <LoginForm />
    </BookStoreProvider>
  );
  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have an email field', () => {
    expect(wrapper.find('Field[name="emailId"]').length).toEqual(1);
  });

  it('should have an password field', () => {
    expect(wrapper.find('Field[name="password"]').length).toEqual(1);
  });

  it('should have proper props for email field', () => {
    expect(wrapper.find('Field[name="emailId"]').props()).toEqual({
      type: 'email',
      placeholder: 'Email Id',
      className: 'form-control',
      name: 'emailId',
      value: "",
      onInput: expect.any(Function),
      onChange: expect.any(Function),
      autoComplete: 'off',
    });
  });

  it('should have proper props for password field', () => {
    expect(wrapper.find('Field[name="password"]').props()).toEqual({
      type: 'password',
      placeholder: 'Password',
      className: 'form-control',
      name: 'password',
      value: '',
      onInput: expect.any(Function),
      onChange: expect.any(Function),
      autoComplete: 'off',
    });
  });

  // it('password field should have new value onchange', () => {
  //   const passwordField = wrapper.find('Field[type="password"]');
  //   passwordField.props().onChange({target: {
  //     name: 'password',
  //     value: '123456',
  //   }});
  //   passwordField.simulate('change');
  //   // wrapper.update();
    // expect(wrapper.find('Field[name="password"]').prop('value')).toEqual(
    //   '123456'
  //   );
  // });

  // it('username field should have new value onchange', (done) => {
  //   const emailField = wrapper.find('Field[name="emailId"]');
  //   console.log(emailField.props().value = 'kakashi.hatake@gmail.com');
  //   emailField.simulate('change', {
  //     target: {
  //       name: 'emailId',
  //       value: 'kakashi.hatake@gmail.com',
  //     },
  //   });
  //   expect(wrapper.find('Field[name="emailId"]').prop('value')).toEqual(
  //     'kakashi.hatake@gmail.com');
  //   done();
  // });

  // it('username field should have new value onchange', (done) => {
  //   // const [testState, setTestState] = useState({
  //   //   loginUserEmailId: '',
  //   //   loginUserPassword: '',
  //   //   isLoggedIn: false,
  //   // });
  //   const contextValues = {
  //     loginUserEmailId: '',
  //     loginUserPassword: '',
  //     isLoggedIn: false
  //   }
  //   jest
  //     .spyOn(BookStoreContext, useBookStoreContext)
  //     .mockImplementation(() => contextValues);
  //   const container = mount(
  //     <BookStoreProvider>
  //       <LoginForm />
  //     </BookStoreProvider>
  //   );
  //   expect(container.find('Field[name="emailId"]').prop('loginUserEmailId')).toBe('')
  //   done();
  // });
});
