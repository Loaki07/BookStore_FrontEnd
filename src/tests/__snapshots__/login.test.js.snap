// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `"<div><form action=\\"#\\"><div class=\\"form-group\\"><label class=\\"form-label\\">Email Id</label><input name=\\"emailId\\" type=\\"email\\" placeholder=\\"Email Id\\" class=\\"form-control\\" autocomplete=\\"off\\" value=\\"\\"></div><div class=\\"form-group\\"><label class=\\"form-label\\">Password</label><input name=\\"password\\" type=\\"password\\" placeholder=\\"Password\\" class=\\"form-control\\" autocomplete=\\"off\\" value=\\"\\"></div><div class=\\"form-group\\"><button type=\\"button\\" class=\\"btn btn-secondary btn-sm\\">Forgot Password?</button></div><div class=\\"form-group\\"><button type=\\"submit\\" class=\\"btn btn-danger btn-block\\">Login</button></div><div class=\\"form-group\\"><span>or</span></div><div class=\\"form-row\\"><div class=\\"form-group\\"><button type=\\"button\\" class=\\"btn btn-info\\">FaceBook</button><button type=\\"button\\" class=\\"btn btn-secondary\\">Google</button></div></div></form></div>"`;
